test_utils/src/lib.rs

```diff
@@ -319,13 +319,23 @@ macro_rules! test_interpreter_and_jit {
         );
     };
     ($executable:expr, $mem:tt, $context_object:expr, $expected_result:expr $(,)?) => {
+        let expected_result = $expected_result;
         test_interpreter_and_jit!(
-            false,
+            override_budget => true,
             $executable,
             $mem,
             $context_object,
-            $expected_result
-        )
+            expected_result,
+        );
+        if !matches!(expected_result, ProgramResult::Err(solana_sbpf::error::EbpfError::ExceededMaxInstructions)) {
+            test_interpreter_and_jit!(
```

Comment:
I don't think it is a good idea to re-run the test with a different configuration if that error comes out. That setting may mask other problems in the code.

I think we should not be overriding the compute budget.

Reply:
It does not re-run it depending on the result.
Instead it either runs one or two versions depending on the **expected** result.
This can only detect more, not less errors.